src/transformStateWithClones.js

```diff
@@ -1,13 +1,42 @@
 'use strict';
 
 /**
- * @param {Object} state
+ * @param {Object} stateCopy
  * @param {Object[]} actions
  *
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArray = [];
+  let stateCopy = { ...state };
+
+  for (const object of actions) {
+    if (object.type === 'addProperties') {
+      const extraData = object.extraData;
+
+      Object.assign(stateCopy, extraData);
```

Comment:
If a constant is used once and does not improve the code in any way (does not make it more universal or more readable), it is not worth creating it
```suggestion
      Object.assign(stateCopy, object.extraData);
```